streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -363,6 +362,98 @@ public synchronized final TopologyBuilder addSource(String name, Deserializer ke
     }
 
     /**
+     * Add a global store to the topology. This store can be accessed by any of the Processors
+     * in the topology.
+     *
+     * @param store  the instance of {@link StateStore}
+     * @return this builder instance so methods can be chained together; never null
+     */
+    public synchronized TopologyBuilder addGlobalStore(final StateStore store) {
```

Comment:
I don't think we should use a suppler here. We want a single instance of a `StateStore` and this shows that intent. A supplier indicates that there may be multiple instances.